otx/api/utils/dataset_utils.py

```diff
@@ -183,3 +189,63 @@ def contains_anomalous_images(dataset: DatasetEntity) -> bool:
         if any(label.is_anomalous for label in labels):
             return True
     return False
+
+
+def add_saliency_maps_to_dataset_item(
+    dataset_item: DatasetItemEntity,
+    saliency_map: np.ndarray,
+    model: Optional[ModelEntity],
+    labels: List[LabelEntity],
+    task: str = "",
+    predicted_scene: AnnotationSceneEntity = None,
+):
+    """Add saliency maps(2d for class-ignore saliency map, 3d for class-wise saliency maps) to a single dataset item.
+
+    Params `task` and `predicted_scene` are only used for openvino task, to select the predicted class' saliency maps.
+    """
+    if saliency_map.ndim == 2:
+        # Single saliency map per image, support e.g. EigenCAM use case
+        actmap = get_actmap(saliency_map, (dataset_item.width, dataset_item.height))
+        saliency_media = ResultMediaEntity(
+            name="Saliency Map",
+            type="saliency_map",
+            annotation_scene=dataset_item.annotation_scene,
+            numpy=actmap,
+            roi=dataset_item.roi,
+        )
+        dataset_item.append_metadata_item(saliency_media, model=model)
+    elif saliency_map.ndim == 3:
+        # Multiple saliency maps per image (class-wise saliency map)
+        # If predicted_scene is provided, add saliency map with only predicted classes(used for openvino task)
+        predicted_labels = set()
+        if predicted_scene is not None:
+            if task == "cls" and len(predicted_scene.annotations) > 0:
+                predicted_labels = set(
+                    scored_label.label for scored_label in predicted_scene.annotations[0].get_labels()
+                )
+            elif task == "det":
+                for bbox in predicted_scene.annotations:
+                    scored_label = bbox.get_labels()[0]
+                    predicted_labels.add(scored_label.label)
+
+        num_saliency_maps = saliency_map.shape[0]
+        if num_saliency_maps == len(labels) + 1:
+            # Include the background as the last category
+            labels.append(LabelEntity("background", Domain.DETECTION))
```

Comment:
This is a common method for cls and det tasks, right?. Can we make some check here (it is better to add this background label only for det task)?